pkg/compactor/blocks_cleaner.go

```diff
@@ -321,15 +326,40 @@ func (c *BlocksCleaner) cleanUser(ctx context.Context, userID string, firstRun b
 		}
 	}
 
+	// Reading bucket index sync stats
+	idxs, err := bucketindex.ReadSyncStatus(ctx, c.bucketClient, userID, userLogger)
+
+	if err != nil {
+		level.Warn(userLogger).Log("msg", "error reading the bucket index status", "err", err)
+		idxs = bucketindex.Status{Version: bucketindex.SyncStatusFileVersion, NonQueryableReason: bucketindex.Unknown}
+	}
+
+	idxs.Status = bucketindex.Ok
+	idxs.SyncTime = time.Now().Unix()
+
 	// Read the bucket index.
 	idx, err := bucketindex.ReadIndex(ctx, c.bucketClient, userID, c.cfgProvider, c.logger)
+
+	defer func() {
+		bucketindex.WriteSyncStatus(ctx, c.bucketClient, userID, idxs, userLogger)
+	}()
+
 	if errors.Is(err, bucketindex.ErrIndexCorrupted) {
 		level.Warn(userLogger).Log("msg", "found a corrupted bucket index, recreating it")
 	} else if errors.Is(err, bucket.ErrCustomerManagedKeyAccessDenied) {
 		// Give up cleaning if we get access denied
-		level.Warn(userLogger).Log("msg", err.Error())
+		level.Warn(userLogger).Log("msg", "customer manager key access denied", "err", err)
+		idxs.Status = bucketindex.CustomerManagedKeyError
+		// Making the tenant non queryable until 3x the cleanup interval to give time to compactors and storegateways
+		// to reload the bucket index in case the key access is re-granted
+		idxs.NonQueryableUntil = time.Now().Add(3 * c.cfg.CleanupInterval).Unix()
```

Comment:
Nit: can we extract the magic number 3 to some meaningfully name constant?

Reply:
done